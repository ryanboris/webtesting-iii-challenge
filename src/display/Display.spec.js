import React from 'react'
import { render, cleanup, fireEvent } from 'react-testing-library'
import Display from './Display'

afterEach(cleanup)

describe('<Display/>', () => {
  test('gate is initially open', () => {
    const { getByTestId } = render(<Display />)
    expect(getByTestId('closed-div').textContent).toBe('Open')
    expect(getByTestId('closed-div')).toMatchSnapshot()
  })

  test('gate is initially unlocked', () => {
    const { getByTestId } = render(<Display />)
    expect(getByTestId('unlocked-div').textContent).toBe('Unlocked')
    expect(getByTestId('unlocked-div')).toMatchSnapshot()
  })
})
