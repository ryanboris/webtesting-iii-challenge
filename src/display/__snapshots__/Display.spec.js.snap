// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display/> gate is initially open 1`] = `
<div
  class="led green-led"
  data-testid="closed-div"
>
  Open
</div>
`;

exports[`<Display/> gate is initially unlocked 1`] = `
<div
  class="led green-led"
  data-testid="unlocked-div"
>
  Unlocked
</div>
`;
